src/main/java/net/datafaker/DateAndTime.java

```diff
@@ -220,15 +239,15 @@ public Date between(Date from, Date to) throws IllegalArgumentException {
      * @throws IllegalArgumentException if the {@code to} date represents an earlier date than {@code from} date.
      */
     public String between(Date from, Date to, String pattern) throws IllegalArgumentException {
-        return toString(between(from, to), pattern);
+        return toString(between(from, to), pattern, faker.getLocale());
     }
 
     /**
      * Generates a random birthday between 65 and 18 years ago from now.
      *
      * @return a random birthday between 65 and 18 years ago from now.
      */
-    public Date birthday() {
+    public LocalDate birthday() {
```

Comment:
I'm not sure about this. This breaks the API. How about instead of changing it here, we have a `DateAndTime` faker, which is the old Java thing, which uses Dates, and a new Java 8 Time, maybe called `Time` or so, which does uses only the Java 8 api, and we make the DateAndTime deprecated? 

I know we don't have many users of this library yet, but ideally I'd make the transition as easy as possible, and keep breaking the API as limited as possible. My own projects are using this library, and it's now going to break things for me.